GCEED/rt/calc_env_trigon.f90

```diff
@@ -31,27 +31,27 @@ subroutine calc_env_trigon(ipulse,tenv_trigon)
 
   if(ipulse==1)then
     tae_shape=ae_shape1
-    ! sin(alpha1*theta1)**2
+    ! cos(alpha1*theta1)**2
     alpha1=Pi/pulse_tw1
-    theta1=dble(itt)*dt
-    ! sin(alpha2*theta2) or cos(alhpa2*theta2)
+    theta1=dble(itt)*dt-0.5d0*pulse_tw1
+    ! cos(alhpa2*theta2)
     alpha2=omega1
     theta2=dble(itt)*dt-0.5d0*pulse_tw1+phi_cep1*2d0*pi
   else if(ipulse==2)then
     tae_shape=ae_shape2
-    ! sin(alpha1*theta1)**2
+    ! cos(alpha1*theta1)**2
     alpha1=Pi/pulse_tw2
-    theta1=dble(itt)*dt-t1_t2
-    ! sin(alpha2*theta2) or cos(alhpa2*theta2)
+    theta1=dble(itt)*dt-t1_t2-0.5d0*pulse_tw2
+    ! cos(alhpa2*theta2)
     alpha2=omega2
     theta2=dble(itt)*dt-t1_t2-0.5d0*pulse_tw2+phi_cep2*2d0*pi
```

Comment:
@mn2007 san, thank you for the implementation. But, it seems that the definition of CEP does not much the conventional one, `cos(w*t+phi)`, since, in your implementation, the CEP depends on the frequency. I mean the argument of the carrier wave contains a product of `alpha2` and `phi_cep2*2d0*pi` in your implementation. However, the CEP part should be separated from the frequency in the conventional definition.